src/FavIconManager.vala

```diff
@@ -36,15 +38,19 @@ public class FeedReader.FavIconManager : GLib.Object {
 		{
 			var feed_id = feed.getFeedID();
 			var future = m_map.get(feed_id);
-			if(future == null)
+
+			bool download_icon = future == null || (future.ready && (future.value.Metadata == null || future.value.Metadata.IsExpired()) );
```

Comment:
I'm not sure why we have a `null` check for `future.value.Metadata`. The `expires` could be null, but the metadata itself shouldn't ever be.

Also `future.value.Metadata.IsExpired()` seems to be an entirely new style of variable naming.. :p